### ISSUE TITLE
accuracy drops a lot in fp16 mode

### ISSUE THREAD

**Issue by itachi1232gg** [NVIDIA-AI-IOT/torch2trt#879]
my model's accuracy drops a lot when I convert it into fp16 mode, even a pretrained resnet34 experienced accuracy drop in fp16 mode.
`
import os  
os.environ['CUDA_MODULE_LOADING'] = 'LAZY'    
from torchvision.models.resnet import resnet34, resnet18    
data = torch.ones((1,3,224,224)).cuda()    
model = resnet18(pretrained=True)    
model.eval()     
model.cuda()     
model_trt = torch2trt(model, [data], fp16_mode=True)     
with torch.no_grad():     
    print((model(data)-model_trt(data)).abs().sum())      
`

> tensor(5.3151, device='cuda:0')   

if I set fp16_mode=False
then the output is 

>tensor(0.0017, device='cuda:0')

**Comment by Lu-tju** [comment]
Hi, the result in my computer is 0.6877 with your code when set fp16_mode=False (and 5.3 when fp16_mode=True). I don't know is this error normal?

**Comment by itachi1232gg** [comment]
> Hi, the result in my computer is 0.6877 with your code when set fp16_mode=False (and 5.3 when fp16_mode=True). I don't know is this error normal?

0.6877 is likely to give you the wrong outputs.

**Comment by JWLee89** [comment]
Instead of testing the absolute sum of differences between the two models, I believe an element-wise difference check to see if all the element-wise differences do not exceed a certain threshold might be a more accurate measure. 

For example (did not test the code), we can check whether all elements inside of source and target tensor are within a certain absolute threshold.

```python
output_pt = model(data)
output_trt = model_trt(data)

# Set this value to something that seems appropriate to you
# 1e-5 is generally reasonable
import numpy as np
absolute_tolerance = 1e-5
np.allclose(output_pt, output_trt, atol=absolute_tolerance)
```

**Comment by itachi1232gg** [comment]
> Instead of testing the absolute sum of differences between the two models, I believe an element-wise difference check to see if all the element-wise differences do not exceed a certain threshold might be a more accurate measure.
> 
> For example (did not test the code), we can check whether all elements inside of source and target tensor are within a certain absolute threshold.
> 
> ```python
> output_pt = model(data)
> output_trt = model_trt(data)
> 
> # Set this value to something that seems appropriate to you
> # 1e-5 is generally reasonable
> import numpy as np
> absolute_tolerance = 1e-5
> np.allclose(output_pt, output_trt, atol=absolute_tolerance)
> ```

np.allclose(output_pt, output_trt, atol=absolute_tolerance) returns False
